src/v/cloud_storage/async_manifest_view.h

```diff
@@ -256,7 +256,8 @@ class async_manifest_view_cursor {
     async_manifest_view_cursor_status get_status() const;
 
     /// Move to the next manifest or fail
-    ss::future<result<bool, error_outcome>> next();
+    using eof = ss::bool_class<struct eof_tag>;
```

Comment:
I think it can return `ss::stop_iteration::yes` because it controls the iteration. This way you can merge `next` and `next_iter`.

Reply:
I don't think they can be merged as `next_iter` also does error checking and throws. I kind of like having a separate type, but I don't mind too much.